src/main/java/io/cryostat/messaging/MessagingServer.java

```diff
@@ -216,6 +221,9 @@ public void start() throws SocketException, UnknownHostException {
     }
 
     public void writeMessage(WsMessage message) {
+        if (message instanceof Notification) {
+            notificationSource.notifyListeners((Notification<?>) message);
```

Comment:
This can be removed now since the `Notification#send()` call is already doing this, so listeners will actually end up getting the notification event twice.